pkg/kapp/config/default.go

```diff
@@ -612,6 +612,11 @@ changeRuleBindings:
           - anyMatcher: {matchers: *serviceAccountMatchers}
           - anyMatcher: {matchers: *rbacMatchers}
 
+- rules:  
+  - "delete after deleting change-groups.kapp.k14s.io/serviceaccount"
```

Comment:
Let's have granular ordering (similar to insertion of namespaced resources) so that we delete a Namespace after the SA only if that SA belongs to that Namespace.
Also, let's add `ignoreIfCyclical` and `disableDefaultChangeGroupAnnMatcher`.